src/GuardClauses/GuardAgainstNullExtensions.cs

```diff
@@ -79,6 +79,29 @@ public static partial class GuardClauseExtensions
             return input;
         }
 
+#if NET5_0_OR_GREATER
+        /// <summary>
+        /// Throws an <see cref="ArgumentException" /> if <paramref name="input" /> is an empty string.
+        /// </summary>
+        /// <param name="guardClause"></param>
+        /// <param name="input"></param>
+        /// <param name="parameterName"></param>
+        /// <param name="message">Optional. Custom error message</param>
+        /// <returns><paramref name="input" /> if the value is not an empty string.</returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static ReadOnlySpan<char> NullOrEmpty(this IGuardClause guardClause,
+            ReadOnlySpan<char> input,
+            string parameterName,
+            string? message = null)
+        {
+            if (input == string.Empty)
```

Comment:
I think this may convert the `Span<char>` to string, perhaps `input.Lenght == 0` is a better fit?

Reply:
Maybe, but I don't see a string conversion in the method signature:
![image](https://user-images.githubusercontent.com/2193940/194927494-7a841be4-a798-484f-90cf-0fad200d7fcb.png)
Is it really important here?

Reply:
Oh nice, I would've expected it to take the most concrete type but it actually just picks the span version here.
I wonder if it matters indeed.
I guess it can be benchmarked but that sounds like diminishing returns to me.

Reply:
I guess functionally it may matter :thinking:.
I had a quick look and a `ReadOnlySpan<char>` and a `System.String` don't seem to have the same equality logic: 

`String` 
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs,750
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs,372d790ddae4cbb4
`Span<T>`
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs,321

However the conversion from string seems pretty optimized so I don't know:
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/String.cs,342

I guess it becomes mostly important if it's generic.

Reply:
So it's better to use `input.Lenght == 0`, right?

Reply:
Maybe copy the approach from here:
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs,321
First check the reference and return fast.
If not then you can always do the equals between the `Span` and `String.Empty`.
Or just call `Unsafe.AreSame<T>(ref left._reference, ref right._reference);` yourself it it's generic.

Reply:
We can't use _reference directly because it is [internal](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs,30)
I can propose this variant:
```
if (input.Length != 0)
{
    return input;
}
if (input == string.Empty)
{
    throw new ArgumentException(message ?? $"Required input {parameterName} was empty.", parameterName);
}
return input;
```

Reply:
Hmm I'd personally go for:
```csharp
if (input.Length == 0 || input == string.Empty)
{
    throw new ArgumentException(message ?? $"Required input {parameterName} was empty.", parameterName);
}
return input;
```
Just so you don't need to do the equals check when the length is 0.